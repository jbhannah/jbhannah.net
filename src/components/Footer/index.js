import React from 'react'
import PropTypes from 'prop-types'
import Link from 'gatsby-link'

const year = (new Date()).getFullYear()

const Footer = ({ title, className }) => (
  <footer className={className}>
    <div>
      Copyright Â© {year}
      {' '}<Link to="/">{title}</Link>.
      {' '}<a href="https://creativecommons.org/license/by-sa/4.0/" target="_blank" rel="license">Some Rights Reserved</a>.
      {' '}<a href="https://github.com/jbhannah/jbhannah.net" target="_blank">Source on GitHub</a>.
    </div>
  </footer>
)

Footer.propTypes = {
  title: PropTypes.string,
}

export default Footer
