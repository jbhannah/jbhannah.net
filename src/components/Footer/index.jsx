import PropTypes from "prop-types"
import React from "react"
import Link from "../Link"

const year = new Date().getFullYear()

const Footer = ({ title, className }) => (
  <footer className={className}>
    <div>
      Copyright Â© {year} <Link href="/">{title}</Link>.{" "}
      <Link
        href="https://creativecommons.org/licenses/by-sa/4.0/"
        target="_blank"
        rel="license"
      >
        Some Rights Reserved
      </Link>
      .{" "}
      <Link href="https://github.com/jbhannah/jbhannah.net" target="_blank">
        Source on GitHub
      </Link>
      .
    </div>
  </footer>
)

Footer.propTypes = {
  title: PropTypes.string,
}

export default Footer
