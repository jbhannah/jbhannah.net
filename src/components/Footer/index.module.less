@import "../../assets/css/_mixins";
@import "../../assets/css/_variables";

.footer {
  &:extend(.content-width-column);

  align-self: flex-end;
  font-size: 0.75rem;
  margin: 1rem auto;

  @media @mq[lg] {
    margin-left: @header-width;
  }

  @media @mq[xl] {
    margin-left: 0;
  }
}
