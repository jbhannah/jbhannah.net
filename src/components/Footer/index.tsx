import styled from "@emotion/styled"
import * as React from "react"
import { contentWidthColumn, headerWidth, mq } from "../../utils/styles"
import { Link } from "../Link"

interface FooterProps {
  fullname: string | null
  pronouns: string | null
}

const year = new Date().getFullYear()

const _Footer = styled.footer({
  ...contentWidthColumn,
  alignSelf: "flex-end",
  fontSize: "0.75rem",
  margin: "1rem auto",
  [mq.lg]: { marginLeft: headerWidth },
  [mq.xl]: { marginLeft: 0 },
})

const Footer: React.FunctionComponent<FooterProps> = ({
  fullname,
  pronouns,
}) => (
  <_Footer>
    <div>
      Copyright Â© {year} <Link href="/">{fullname}</Link> ({pronouns}).{" "}
      <Link
        href="https://creativecommons.org/licenses/by-sa/4.0/"
        rel="license"
      >
        Some Rights Reserved
      </Link>
      . Emoji by <Link href="https://twemoji.twitter.com/">Twemoji</Link>.
      <Link href="https://github.com/jbhannah/jbhannah.net">
        Source on GitHub
      </Link>
      .
    </div>
  </_Footer>
)

export default Footer
