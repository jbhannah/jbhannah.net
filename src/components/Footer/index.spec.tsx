import { shallow } from "enzyme"
import MockDate from "mockdate"
import * as React from "react"
import Footer from "."

MockDate.set("2019-02-05T01:17:07.551Z")

const TestFooter = (props) => shallow(<Footer {...props} />)

describe("Footer", () => {
  const tree = TestFooter({ fullname: "Fullname", pronouns: "Pronouns" })

  it("renderes correctly", () => {
    expect(tree).toMatchSnapshot()
  })
})
