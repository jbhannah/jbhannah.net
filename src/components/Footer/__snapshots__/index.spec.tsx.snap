// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renderes correctly 1`] = `
<Styled(footer)>
  <div>
    Copyright Â© 
    2021
     
    <Link
      href="/"
    >
      Fullname
    </Link>
     (
    Pronouns
    ).
     
    <Link
      href="https://creativecommons.org/licenses/by-sa/4.0/"
      rel="license"
    >
      Some Rights Reserved
    </Link>
    .
     
    <Link
      href="https://github.com/jbhannah/jbhannah.net"
    >
      Source on GitHub
    </Link>
    .
  </div>
</Styled(footer)>
`;
